src/Service.Tests/SqlTests/GraphQLMutationTests/GraphQLMutationTestBase.cs

```diff
@@ -206,6 +206,100 @@ public async Task TestStoredProcedureMutationForDeletion(string dbQueryToVerifyD
             Assert.AreEqual(updatedResult.RootElement.GetProperty("maxId").GetInt64(), 13);
         }
 
+        public async Task InsertMutationOnTableWithTriggerWithNonAutoGenPK(string dbQuery)
+        {
+            // Validate that insert operation succeeds when an insert DML trigger is enabled for a table with non-autogenerated primary key.
+            // For non-autogenerated PK fields, the values are inserted into a temporary table and then an INNER JOIN is performed
+            // based on those fields' values from the actual table and the temporary table.
+            // Trigger behavior: The trigger updates the salary to max(0,min(30,salary)).
+            // We would not need to specify the value of the PK column 'months' as it has a default value of 2.
+            string graphQLMutationName = "createInternData";
+            string graphQLMutation = @"
+                mutation{
+                    createInternData(item: { id: 4, name: ""Tommy"", salary: 45 }) {
+                        id
+                        months
+                        name
+                        salary
+                    }
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, isAuthenticated: true, new() { { "title", "My New Book" }, { "publisher_id", 1234 } });
```

Comment:
How does this test past as-is when variables unrelated to `createInternData` are provided?

`{ "title", "My New Book" }, { "publisher_id", 1234 }`

Reply:
this is present for the next handfull of tests. If it is as simple as removing, can you follow up with why their presence didn't fail the tests?

Reply:
The variables were not referenced anywhere in the gql mutation and hence their absence/presence does not make any difference. Thanks for catching!